st2common/st2common/util/action_db.py

```diff
@@ -193,6 +194,22 @@ def update_liveaction_status(status=None, result=None, context=None, end_timesta
                          'to unknown status string. Unknown status is "%s"',
                          liveaction_db, status)
 
+    if liveaction_db.status:
```

Comment:
Maybe a comment would make it a bit easier to understand what is going on - "decrement the counter for current / previous" status here and below "increment the counter for current / new status" or similar.